File: backend/src/main/java/com/smartcampus/incident/service/AuthService.java
package com.smartcampus.incident.service;

import com.smartcampus.incident.dto.auth.AuthResponse;
import com.smartcampus.incident.dto.auth.LoginRequest;
import com.smartcampus.incident.dto.auth.RegisterRequest;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);

    String verifyOtp(String email, String otp);
    String resendOtp(String email);

    void changePassword(com.smartcampus.incident.entity.User currentUser, com.smartcampus.incident.dto.user.ChangePasswordRequest request);
}

File: frontend/src/api/authService.js
import api from './axiosInstance'

export const authService = {
  login: (data) => api.post('/auth/login', data),
  register: (data) => api.post('/auth/register', data),

  //otp features
  verifyOtp: (data) => api.post('/auth/verify-otp', data),
  resendOtp: (email) => api.post('/auth/resend-otp', { email }),
  
  getMe: () => api.get('/users/me'),
  updateProfile: (data) => api.put('/users/me', data),
  changePassword: (data) => api.put('/users/me/password', data),
}
